util/dvsim/LocalLauncher.py

```diff
@@ -74,6 +84,9 @@ def poll(self):
         '''
 
         assert self.process is not None
+        if self.timeout_secs:
+            if self.start_time + self.timeout_secs < datetime.datetime.now():
+                self.kill()
```

Comment:
Should we move this inside the `if self.process.poll() is None:` clause? That way, if the timeout occurred but the process also completed successfully in the same polling step, the timeout expiry is ignored. 

Also consider refactoring into its own method, such as `self.kill_if_timeout_exceeded()` (also see next comment below).

Reply:
Done